### PR TITLE
[Enhancement](icebergv2) Optimize the position delete file filtering mechanism in iceberg v2 parquet reader.

### LINKED ISSUE #16023: [Enhancement] Optimize the position delete file filtering mechanism in iceberg v2.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, after obtaining the delete position through Iceberg position delete files, the parquet reader adopts a mechanism of dividing the delete position into different row ranges to filter. When there are too many delete positions, the number of virtual function calls to read decoded columns will increase, and `column_data.resize()` will be called too many times。

parquet_common.h
```
template <typename Numeric>
Status FixLengthDecoder::_decode_numeric(MutableColumnPtr& doris_column,
                                         ColumnSelectVector& select_vector) {
    auto& column_data = static_cast<ColumnVector<Numeric>&>(*doris_column).get_data();
    size_t data_index = column_data.size();
    column_data.resize(data_index + select_vector.num_values() - select_vector.num_filtered());
    ...
```

### Solution

Merge delete position filter with condition filter to handle it.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kaka11chen**
# Proposed changes

Issue Number: close #16023

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/format/parquet/vparquet_group_reader.cpp [hunk 1]
```diff
@@ -262,8 +258,9 @@ Status RowGroupReader::_do_lazy_read(Block* block, size_t batch_size, size_t* re
             // generated from next batch, so the filter column is removed ahead.
             DCHECK_EQ(block->rows(), 0);
         } else {
-            Block::filter_block(block, _lazy_read_ctx.all_predicate_col_ids, filter_column_id,
-                                origin_column_num);
+            ColumnPtr filter_column = block->get_by_position(filter_column_id).column;
```
Comment (AshinGau) [inline be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:261]: We should filter the predicate columns by `_filter_ptr` when there're delete rows.

File: be/src/vec/exec/format/parquet/vparquet_group_reader.cpp [hunk 2]
```diff
@@ -434,6 +438,121 @@ Status RowGroupReader::_read_empty_batch(size_t batch_size, size_t* read_rows, b
...
+    if (!_position_delete_ctx.has_filter) {
+        return Status::OK();
+    }
+    size_t remaining_read_rows = _total_read_rows + read_rows;
+    while (_position_delete_ctx.index < _position_delete_ctx.end_index) {
+        const int64_t delete_row_index_in_row_group =
+                _position_delete_ctx.delete_rows[_position_delete_ctx.index] -
+                _position_delete_ctx.first_row_id;
+        int64_t read_range_rows = 0;
+        for (auto& range : _read_ranges) {
```
Comment (AshinGau) [inline be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:453]: The two-level loop is time-consuming.

File: be/src/vec/exec/format/parquet/vparquet_group_reader.cpp [hunk 3]
```diff
@@ -434,6 +438,121 @@ Status RowGroupReader::_read_empty_batch(size_t batch_size, size_t* read_rows, b
...
+            if (delete_row_index_in_row_group < range.first_row) {
+                break;
+            } else if (delete_row_index_in_row_group < range.last_row) {
```
Comment (AshinGau) [inline be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:456]: Maybe we can build an inline function to calculate the line number

File: be/src/vec/exec/format/parquet/vparquet_group_reader.cpp (hunk 1 above)
Comment (kaka11chen) [reply be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:261]: Filter by _filter_ptr in the _filter_block function.

**Comment by AshinGau** [comment]
LGTM

File: be/src/vec/exec/format/parquet/vparquet_group_reader.cpp (hunk 2 above)
Comment (kaka11chen) [reply be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:453]: Don't search next range when there is no remaining_read_rows.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**